Source/Tools/LinuxEmulation/LinuxSyscalls/GdbServer.cpp

```diff
@@ -1426,13 +1428,17 @@ void GdbServer::GdbServerLoop() {
         SendPacketPair({std::move(str), HandledPacketType::TYPE_ACK});
         break;
       }
-      default: LogMan::Msg::DFmt("GdbServer: Unexpected byte {} ({:02x})", static_cast<char>(c), c);
+      default: LogMan::Msg::DFmt("GdbServer: Unexpected byte {} ({:02x})", c.GetData(), c.GetData());
       }
     }
 
+    if (c.HasHangup()) {
```

Comment:
The control flow has become weird here (maybe since the latest changes?).

Why do we skip the `InvalidateSocket` call when we're receiving a Hangup?

Why is there still an outer `while (!CoreShuttingDown.load()) {` loop? I don't see how that outer loop would ever run more than one iteration, since we reach this line only when a `Hangup` is received or `CoreShuttingDown==true`.

Reply:
Yep, that was a bug introduced due to the refactoring request to delete an outer loop. Added the loop back because it's required to allow gdb connections to disconnect and reconnect at a later time.

Reply:
There's *three* levels of nesting loops now, two of them on the same variable...

```cpp
  while (!CoreShuttingDown.load()) {
    // ...

    while (!CoreShuttingDown.load()) {
      // ...
      while ((c = CommsStream.get()).HasData()) {
```